### PR TITLE
CSV Staging Extractor

### PR DESCRIPTION
**Pull request by danlee1025**
# Summary
- Added `CSVStagingExtractor` and is exported from MEF
## New behavior & Code changes
- Added `TNMCategoryTemplate`, `StagingTemplate` and `CSVStagingExtractor`
    - The `get` function in this extractor is a little different because I had to format and generate the resources for each category(t, n, m) and then pass the ids to the staging template so that it could be referenced correctly.
- Added tests for new templates and extractor
# Testing guidance
- Ensure tests pass

### CODE REVIEW

File: src/extractors/CSVStagingExtractor.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+const { firstEntryInBundle } = require('../helpers/fhirUtils');
+const { generateMcodeResources } = require('../templates');
+const logger = require('../helpers/logger');
+const { formatDateTime } = require('../helpers/dateUtils');
+
+function formatTNMCategoryData(stagingData) {
+  logger.debug('Reformatting TNM Category data into template format');
+  const formattedData = [];
+  const {
+    mrn, conditionId, t, m, n, type, effectiveDate,
```
Comment (Dtphelan1) [inline src/extractors/CSVStagingExtractor.js:13]: petty, almost irrelevant suggestion that would make me disproportionately happy: reordering `t` `m` and  `n` to `t`, `n`,  and `m`
```suggestion
    mrn, conditionId, t, n, m, type, effectiveDate,
```

File: test/extractors/CSVStagingExtractor.test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+describe('CSVStagingExtractor', () => {
+  describe('formatTNMCategoryData', () => {
+    test('should join data appropriately and throw errors when missing required properties', () => {
+      const expectedErrorString = 'Staging data is missing an expected property: mrn, conditionId, effectiveDate are required.';
+      const localData = _.cloneDeep(exampleStagingModuleResponse[0]);
+      // Test that valid data works fine
+      expect(formatTNMCategoryData(localData)).toEqual(expect.anything());
+
+      // Test all required properties are
```
Comment (Dtphelan1) [inline test/extractors/CSVStagingExtractor.test.js:31]: Could we elaborate on this a bit more? 
```suggestion
      // Test all optional properties can be removed without throwing errors
```

**Comment by Dtphelan1** [comment]
Recording that I was able to rebase this ontop of the standalone-mCODE-client branch and confirm that it integrates with no issue. Also works as expected when 't', 'n', 'm' elements are missing from a row instance. Things are looking good 👍

File: test/extractors/CSVStagingExtractor.test.js (hunk 2 above)
Comment (danlee1025) [reply test/extractors/CSVStagingExtractor.test.js:31]: ashamed to say this is straight copy and paste 😂 , will elaborate!

**Review by julianxcarter — APPROVED** [review]
I tested this using the base-icare client rather than the MEF client that @Dtphelan1 recently pioneered, and I can confirm that it works as expected using that client structure as well! The tests are passing and nothing is jumping out to me concerning the code, so I'm gonna go ahead and approve this. Nice work Dan!

**Comment by Dtphelan1** [comment]
With those small changes I think this looks good to me. Nice work Dan.

**Review by Dtphelan1 — APPROVED** [review]

**Pull request merged**